Add archive button to classroom settings

EricPickup opened the pull request:
## What
Related to #602 and #1996. Migration separated in #2028.

This PR adds a button to the classroom settings page that will toggle whether the classroom is archived.

This will change the `archived_at` attribute in `Organization`.

![image](https://user-images.githubusercontent.com/1490434/61418753-6264e280-a8b0-11e9-92ca-afa92bf5001b.png)

![image](https://user-images.githubusercontent.com/1490434/61418764-714b9500-a8b0-11e9-9adf-4e6a3d2467ae.png)

d12 commented:
Nice, this is super simple! 🤘 

The way this danger zone was designed in the past isn't looking too great with multiple buttons. What do you think about layout out the text/buttons similarly to how we do it on GitHub.com?

<img width="743" alt="Screen Shot 2019-07-18 at 9 18 00 AM" src="https://user-images.githubusercontent.com/30920216/61460649-26c42a00-a93d-11e9-8abf-149e3e9c5a75.png">

d12 commented:
Also can we specify that this won't _delete_ anything?

EricPickup commented:
@d12 I've updated it with the GitHub.com style and added separate titles/descriptions for archived vs unarchived classrooms.
![image](https://user-images.githubusercontent.com/1490434/61481652-94785200-a94d-11e9-9990-45776243ec8e.png)

![image](https://user-images.githubusercontent.com/1490434/61481685-a22dd780-a94d-11e9-8fb4-63aab293e07c.png)

I also moved this button under a feature flag since it doesn't actually do anything user-facing yet.
